src/icesat2_tracks/app.py

```diff
@@ -36,7 +41,38 @@
     validate_track_name_steps_gt_1,
 )
 
+_logger = logging.getLogger(__name__)
+
 app = Typer(add_completion=False)
+
+
+class _LogLevel(str, Enum):
+    QUIET = "quiet"
+    WARNING = "warning"
+    VERBOSE = "verbose"
+    DEBUG = "debug"
+
+
+_LOG_LEVEL_MAPPING = {
+    _LogLevel.QUIET: logging.CRITICAL,
+    _LogLevel.WARNING: logging.WARNING,
+    _LogLevel.VERBOSE: logging.INFO,
+    _LogLevel.DEBUG: logging.DEBUG,
+}
+
+
+@app.callback()
+def main(
+    log: Annotated[
+        _LogLevel,
+        Option(help="Set the log level"),
+    ] = "warning"
+):
+    """Analyze data from the ICESAT2 satellite."""
+    level = _LOG_LEVEL_MAPPING[log]
+    logging.basicConfig(level=level)
```

Comment:
I like the flexibility of this design. What do you think about embedding the mapping as a property of the enum and use the level directly in `logging.basicConfig`?
```suggestion
class _LogLevel(str, Enum):
    QUIET = "quiet"
    WARNING = "warning"
    VERBOSE = "verbose"
    DEBUG = "debug"

    @property
    def level(self):
        return {
            _LogLevel.QUIET: logging.CRITICAL,
            _LogLevel.WARNING: logging.WARNING,
            _LogLevel.VERBOSE: logging.INFO,
            _LogLevel.DEBUG: logging.DEBUG,
        }[self.value]
 
       
@app.callback()
def main(
    log: Annotated[
        _LogLevel,
        Option(help="Set the log level"),
    ] = "warning"
):
    """Analyze data from the ICESAT2 satellite."""
    logging.basicConfig(level=log.level)
```

This avoids using the dict as a global in `main`.

Reply:
Nice! Let's do that.